### PR TITLE
Allow GraphQL::Analysis::AST::FieldUsage to detect use of deprecated enum value as argument

### PR DESCRIPTION
**Pull request by lukes**
Hi, @rmosolgo! What do you think about this change? Thank you!

### CODE REVIEW

**Comment by lukes** [comment]
I noticed there isn't support for input objects for mutations, but wasn't able to work out how to fix that so I've raised an issue: https://github.com/rmosolgo/graphql-ruby/issues/4805. 🙏

File: spec/graphql/analysis/ast/field_usage_spec.rb [hunk 1]
```diff
@@ -138,6 +138,20 @@
...
+    let(:query_string) {%|
+      query {
+        fromSource(source: YAK) {
+          id
+        }
+      }
+    |}
+
+    it "keeps track of deprecated arguments" do
+      assert_equal ['Query.fromSource.source'], result[:used_deprecated_arguments]
```
Comment (lukes) [inline spec/graphql/analysis/ast/field_usage_spec.rb:151]: The interesting thing is `Query.fromSource.source` is the argument name, but it's the argument value `YAK` in this case that's deprecated. I wonder if there's a path we should return that is more explicit about the problem?

**Comment by rmosolgo** [comment]
I pushed a change that: 

- Added a new Set for deprecated enum values 
- Improved handling of non-null and list argument values 
- And after that, the example from #4805 worked 🎉

**Comment by lukes** [comment]
@rmosolgo Thank you very much 🙏!

**Pull request merged**